### ISSUE TITLE
Benchmark list is failed to be displayed in GUI

### ISSUE THREAD

**Issue by clin99** [hobbit-project/platform#245]
## Problem
The online platform fails to load the benchmark list in GUI. 

## Cause
The controller is not answering the GUI, anymore. This is caused by a single request that creates a deadlock.

```
"pool-1-thread-65" #1943 prio=5 os_prio=0 tid=0x00007f2c5c03d000 nid=0x78c waiting on condition [0x00007f2cce0f0000]
   java.lang.Thread.State: WAITING (parking)
	at sun.misc.Unsafe.park(Native Method)
	- parking to wait for  <0x00000005cb62ba58> (a java.util.concurrent.Semaphore$NonfairSync)
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt(AbstractQueuedSynchronizer.java:836)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireSharedInterruptibly(AbstractQueuedSynchronizer.java:997)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly(AbstractQueuedSynchronizer.java:1304)
	at java.util.concurrent.Semaphore.acquire(Semaphore.java:312)
	at org.hobbit.controller.ExperimentManager.stopExperimentIfRunning(ExperimentManager.java:711)
	at org.hobbit.controller.PlatformController.handleFrontEndCmd(PlatformController.java:607)
	at org.hobbit.controller.PlatformController$1.handleDelivery(PlatformController.java:209)
	at com.rabbitmq.client.impl.ConsumerDispatcher$5.run(ConsumerDispatcher.java:144)
	at com.rabbitmq.client.impl.ConsumerWorkService$WorkPoolRunnable.run(ConsumerWorkService.java:99)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

The complete dump of the VM can be found at [hobbit-dump.txt](https://github.com/hobbit-project/platform/files/1848696/hobbit-dump.txt).

## Solution
- [x] Resolve the source for the deadlock
- [x] Make sure that controller is able to answer multiple requests in parallel

**Comment by MichaelRoeder** [comment]
I updated your issue.

Please note that because of maintenance of servers and server rooms, we can not update the platform before Wednesday.

**Comment by MichaelRoeder** [comment]
From what I gathered so far the source of the deadlock is #243.
1. The platform starts a new experiment
    1. It acquires the `experimentMutex` to be able to write to the experiment state
    2. It is creating a new experiment state object and prepares the experiment
    3. It prefetches images and #243 occurs
2. A user wants to stop a running experiment or remove the an experiment from the queue
    1. The controller handles the message coming from the GUI
    2. It removes the experiment from the queue 
    3. It calls `stopExperimentIfRunning` to stop the running experiment if it is the experiment that should be removed.
    4. The method tries to acquire the `experimentMutex` which is not possible because it is still owned by the experiment creation thread which got stuck

**Comment by MichaelRoeder** [comment]
Extended the message handling in the controller to work with more than one incoming message. However, testing is blocked by #243

**Comment by MichaelRoeder** [comment]
Fixed in v2.0.2
